### ISSUE TITLE
script permissions

### ISSUE THREAD

**Issue by DWomack** [jrsupplee/mailinabox#4]
John, your instructions for installing 0.41 quota over 0.41 say to run setup/bootstrap.sh

I had to chmod 777 setup/bootstrap.sh to give it execute permission to run.

Running this now in AWS Lightsail.  Cheap.  40gig SSD for $5/mo.

Dennis

**Comment by jrsupplee** [comment]
You had to set execute permissions because you executed the script directly.  If you pass the script to `bash` like in the instructions, it does not have to be executable.

**Comment by DWomack** [comment]
0.41 to 0.41 quota beta

From the GitHub instructions:

cd into mailinabox and run sudo setup/bootstrap.sh On occasion there are lock errors when updating Munin. Just re-run sudo setup/start.sh until the error does not occur.


Not complaining, just letting you know. 


Dennis

**Comment by jrsupplee** [comment]
I see the line in the README without the `bash` now.  I fixed the README to be `sudo bash ...`

**Comment by DWomack** [comment]
Have you tried .42b with your mods?

Just wondering. 

I have had zero issues with your quota mods. 

Keep me in the loop. 


Dennis

**Comment by ctrl-i** [comment]
I have been using the quota mod without any problems and have upgraded to 0.42b.  The quota mod is excellent and should be added to any mailinabox installation.

**Comment by DWomack** [comment]
Done this yet?

Looks like Josh has posted version 0.42b on mailinabox.email. 


Dennis

Sent from my iPad

**Comment by alento-group** [comment]
> I have been using the quota mod without any problems and have upgraded to 0.42b. The quota mod is excellent and should be added to any mailinabox installation.

So are you saying that Josh's version can be installed over top the quota version and the quota's will be retained?  Now sure how you upgraded to 0.42b - can you please clarify?

**Comment by jrsupplee** [comment]
Installing over top is not recommended.  I have upgraded the repository to include the changes in v0.42b

**Comment by stylnchris** [comment]
When you say installing over the top is no recommended, I'm guessing you
mean over the top of an existing 0.41-quota?

What are the concerns?  Are the scripts not able to understand that there
is a existing quota set?

**Comment by alento-group** [comment]
> When you say installing over the top is no recommended, I'm guessing you mean over the top of an existing 0.41-quota?

A previous poster indicated that they installed v0.42b over top of the 0.41-quota ... exactly what issues would arise is not clear but that is very dangerous IMHO as the database structure would be different at a minimum.  These are two different forks and should not be intermixed.

**Comment by ctrl-i** [comment]
> > When you say installing over the top is no recommended, I'm guessing you mean over the top of an existing 0.41-quota?
> 
> A previous poster indicated that they installed v0.42b over top of the 0.41-quota ... exactly what issues would arise is not clear but that is very dangerous IMHO as the database structure would be different at a minimum. These are two different forks and should not be intermixed.

The previous poster did not say they installed 0.42b over the top.

What are the database structure differences that you speak of?  The process I used was not very dangerous at all.

**Comment by alento-group** [comment]
> I have been using the quota mod without any problems and have upgraded to 0.42b. The quota mod is excellent and should be added to any mailinabox installation.

The only way that you could have upgraded to 0.42b would be by installing it over the top as since the date you wrote that v0.42b-quota did not exist yet.  Your own words state that you have been using the quota mod and have upgraded to v0.42, which would be impossible unless you installed it over the top ... unless you were being off topic and discussing two different installs, which if were the case I'd have to wonder why you even brought it up?  So ... anyways.

If you had followed John's discourse discussion on this subject from the beginning as I have, you would know that he modified the database structure to hold the additional fields required to handle quota's.  For specifics, you would need to ask him directly.

**Comment by ctrl-i** [comment]
> > I have been using the quota mod without any problems and have upgraded to 0.42b. The quota mod is excellent and should be added to any mailinabox installation.
> 
> The only way that you could have upgraded to 0.42b would be by installing it over the top as since the date you wrote that v0.42b-quota did not exist yet. Your own words state that you have been using the quota mod and have upgraded to v0.42, which would be impossible unless you installed it over the top ... unless you were being off topic and discussing two different installs, which if were the case I'd have to wonder why you even brought it up? So ... anyways.
> 
> If you had followed John's discourse discussion on this subject from the beginning as I have, you would know that he modified the database structure to hold the additional fields required to handle quota's. For specifics, you would need to ask him directly.

I bow to your superior knowledge of what I did.

**Comment by jrsupplee** [comment]
> What are the concerns? Are the scripts not able to understand that there is a existing quota set?
> [](#)

It really depends on how you are doing it.  The `mailinabox` script upgrades by checking the latest version on github and then doing a `git pull` to the latest version.  If you have made any changes to the code then the `git pull` might fail because of conflicts and thus the upgrade fails.

There are no data structure differences in the quota version except for the addition of a user quota amount to the SQLite user database.

**Comment by DWomack** [comment]
I have 3 boxes running 0.41 quota, 2 on Linode and 1 on Lightsail. 

The one on Lightsail has next to no use because sending to it from Apple Mail bounces back. The Apple Mail is detecting a blacklist used by Apple.  I have tried multiple times to get the blacklist vendor to fix the problem, to no avail. 

I may spin up another Lightsail instance. Josh says MiaB needs 500 meg of memory now.  A Lightsail with 500 meg is just $3/mo. 

I haven’t touched any code other than upgrading the Linux with apt-get update and apt-get upgrade. 

Is it safe for me to install 0.42b quota over 0.41 quota?

Sorry to be so paranoid. 


Dennis

**Comment by jrsupplee** [comment]
Nothing has changed in the quota code between the two versions.  It should be fine to upgrade.  I had no problems

**Comment by DWomack** [comment]
Upgrade on the linodes went fine. Had to provision new certificates for the added sub domains. 

The Lightsail needed a git stash before git pull for some reason.  Need to configure the added sub domains in the Lightsail DNS records. 

Thanks for the help. 


Dennis

**Comment by DWomack** [comment]
Wow, just as soon as I upgrade, Josh releases 0.43. 

No rush.  


Dennis

**Comment by DWomack** [comment]
You labored on Labor Day!

I just upgraded all three of my email servers to 0.43 with quotas without issues.

Thanks, John.


Dennis

**Comment by stylnchris** [comment]
I am reporting no issues with 43-quota just upgraded not to long ago and it
seems to be working fine.

Zpush - OK
RoundCube - OK
NextCloud - OK
Admin Panel - OK
Web server - OK


Thanks All

**Comment by DWomack** [comment]
Hi John

Just letting you I updated all three of my servers to your latest code. 

The one on Lightsail went fine.  No issues. 

The first one on Linode didn't complete because of lock issues with SpamAss...

Ran the bootstrap.sh again and it locked in another place. 

Updated the second Linode without issue. 

I went back tried to update the first Linode a third time and it completed without issue. 

I grabbed a screen capture of the lock message, If you want to see it. 

Thanks for your efforts with Quotas. 


Dennis
